core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -533,6 +596,12 @@ static boolean shouldExecuteReplicaItem(final BulkItemRequest request, final int
         return replica.delete(delete);
     }
 
+    private static Engine.NoOpResult executeFailedSeqNoOnReplica(BulkItemResponse.Failure primaryFailure, DocWriteRequest docWriteRequest, IndexShard replica) throws IOException {
```

Comment:
I think this should be named something a little closer to what it does (the name makes it sound like it executes the failing request on the replica), so maybe `executeFailureNoOpOnReplica(..)`?